src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,39 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  static alive = [];
+
+  constructor(name, health = 100) {
+    this.name = name;
+    this.health = health;
+    this.hidden = false;
+    Animal.alive.push(this);
+  }
+
+  static removeDeadAnimals() {
+    Animal.alive = Animal.alive.filter((animal) => animal.health > 0);
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(animal) {
+    if (!(animal instanceof Animal) || animal.hidden) {
+      return;
+    }
+
+    if (animal instanceof Carnivore) {
+      return;
+    }
+
+    animal.health -= 50;
+    Animal.removeDeadAnimals();
+  }
```

Comment:
```suggestion
    if (animal instanceof Herbivore && !animal.hidden) {
      animal.health -= 50;
      Animal.removeDeadAnimals();
    }
```

Let's simplify this comdition